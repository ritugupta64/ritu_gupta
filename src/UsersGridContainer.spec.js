import React from "react";
import { shallow } from "enzyme";
import UsersGridContainer from "./usersGridContainer";

describe('UsersGridContainer test', () => {
  const  testProp = {
    key: 4,
    firstName: "Eve",
    lastName: "Holt",
    avatar:"https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
    delete:jest.fn(() => ({}))
  };
  
  it('should behave like tree : UsersGridContainer', () => {
    const wrapper=shallow(
        <UsersGridContainer {...testProp} />
    );

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.length).toBe(1);
    console.info(wrapper.html());
    const h3 = wrapper.find("h3");
    expect(h3.text()).toEqual("Eve Holt ");
     });
  
});