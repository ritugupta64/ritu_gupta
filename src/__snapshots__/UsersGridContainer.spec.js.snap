// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersGridContainer test should behave like tree : UsersGridContainer 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UsersGridContainer
    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
    delete={[Function]}
    firstName="Eve"
    lastName="Holt"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="radius-conatiner"
>
          <img
                    alt="Eve"
                    className="user-thumbnail"
                    src="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
          />
</div>,
        <h3
          className="userName"
>
          Eve
           
          Holt
           
</h3>,
        <span
          className="delete"
          onClick={[Function]}
>
          Delete
</span>,
      ],
      "className": "user",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="Eve"
            className="user-thumbnail"
            src="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
/>,
          "className": "radius-conatiner",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Eve",
            "className": "user-thumbnail",
            "src": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Eve",
            " ",
            "Holt",
            " ",
          ],
          "className": "userName",
        },
        "ref": null,
        "rendered": Array [
          "Eve",
          " ",
          "Holt",
          " ",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "className": "delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="radius-conatiner"
>
            <img
                        alt="Eve"
                        className="user-thumbnail"
                        src="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
            />
</div>,
          <h3
            className="userName"
>
            Eve
             
            Holt
             
</h3>,
          <span
            className="delete"
            onClick={[Function]}
>
            Delete
</span>,
        ],
        "className": "user",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="Eve"
              className="user-thumbnail"
              src="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
/>,
            "className": "radius-conatiner",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Eve",
              "className": "user-thumbnail",
              "src": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Eve",
              " ",
              "Holt",
              " ",
            ],
            "className": "userName",
          },
          "ref": null,
          "rendered": Array [
            "Eve",
            " ",
            "Holt",
            " ",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "className": "delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
